src/main/kotlin/no/nav/familie/ef/sak/behandlingsflyt/steg/BeregnYtelseSteg.kt

```diff
@@ -62,6 +64,22 @@ class BeregnYtelseSteg(private val tilkjentYtelseService: TilkjentYtelseService,
         }
     }
 
+    private fun validerGyldigeVedtaksperioder(behandling: Behandling, data: VedtakDto) {
+        if (data is Innvilget) {
+            val harOpphørsperioder = data.perioder.any { it.periodeType == VedtaksperiodeType.MIDLERTIDIG_OPPHØR }
+            val harInnvilgedePerioder = data.perioder.any { it.periodeType != VedtaksperiodeType.MIDLERTIDIG_OPPHØR }
+            feilHvis(harOpphørsperioder && !harInnvilgedePerioder) {
+                "Må ha innvilgelsesperioder i tillegg til opphørsperioder"
+            }
+        }
+        if (data is Innvilget && data.resultatType == ResultatType.INNVILGE_MED_OPPHØR) {
```

Comment:
Var `INNVILGE_MED_OPPHØR` for å enklere kunne feturetoggle den nye funksjonaliteten?

Reply:
Nei, det er et funksjonelt krav at det skal være et eget valg i nedtrekkslisten/historisk visning.